src/constructs/loadbalancing/alb/application-listener.test.ts

```diff
@@ -165,69 +178,23 @@ describe("The GuHttpsApplicationListener class", () => {
     });
   });
 
-  test("creates certificate prop if no value passed in", () => {
+  test("wires up the certificate which is passed in correctly", () => {
     const stack = simpleGuStackForTesting();
 
     new GuHttpsApplicationListener(stack, "ApplicationListener", {
+      certificate: getCertificate(stack),
       loadBalancer: getLoadBalancer(stack),
       targetGroup: getAppTargetGroup(stack),
       ...app,
     });
-
-    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
-
-    expect(json.Parameters.TLSCertificateTesting).toEqual({
-      Type: "String",
-      AllowedPattern: RegexPattern.ACM_ARN,
-      ConstraintDescription: "Must be an ACM ARN resource",
-      Description: "The ARN of an ACM certificate for use on a load balancer",
-    });
-
     expect(stack).toHaveResource("AWS::ElasticLoadBalancingV2::Listener", {
       Certificates: [
         {
           CertificateArn: {
-            Ref: "TLSCertificateTesting",
+            Ref: "CertificateTesting28FCAC6D",
```

Comment:
Might be overkill, should we test the cert is added to the stack too?

Reply:
I think this would probably be overkill, since adding the cert to the stack is not the responsibility of this construct. We [test cert creation](https://github.com/guardian/cdk/blob/314af33330c2445c4d8de779dbedd9ea6507bffe/src/constructs/acm/certificate.test.ts) as part of the tests for the `GuCertificate` construct, so I think it's probably cleaner to keep that separation of concerns here.